.github/workflows/server_ci.yml

```diff
@@ -1,8 +1,6 @@
 on:
   workflow_dispatch:
   push:
-    branches:
-      - main
```

Comment:
can we make that on pull_request? and switch this be in the tools ci so the tools image is correctly built before it's needed?